Module#02/service/src/main/java/com/epam/esm/service/DefaultCertificateService.java

```diff
@@ -0,0 +1,81 @@
+package com.epam.esm.service;
+
+import com.epam.esm.criteria.Criteria;
+import com.epam.esm.dao.CertificateDao;
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.dto.CertificateWithoutTagDto;
+import com.epam.esm.exception.CertificateIsExistsException;
+import com.epam.esm.exception.CertificateNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+import static com.epam.esm.mapper.CertificateMapper.mapper;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class DefaultCertificateService implements CertificateService {
+
+    private final CertificateDao certificateDao;
+
+    @Override
+    @Transactional(readOnly = true)
+    public CertificateDto getById(final Long id) {
+        return mapper.toDto(certificateDao.getById(id));
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<CertificateDto> getAll() {
+        return mapper.toDtoList(certificateDao.getAll());
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public CertificateDto getByName(final String name) {
+        return mapper.toDto(certificateDao.getByName(name));
+    }
+
+    @Override
+    @Transactional
+    public boolean delete(final Long id) {
+        return certificateDao.delete(id);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<CertificateDto> getAllBy(final Criteria criteria) {
+        return mapper.toDtoList(certificateDao.getAllBy(criteria));
+    }
+
+    @Override
+    @Transactional
+    public boolean update(final CertificateDto certificateDto) {
```

Comment:
There is no validation on null or blank values in fields in ...Dto.

Try to use @Valid annotation

Links:
- https://www.youtube.com/watch?v=d3DANVuyl3M - is good example
- https://www.baeldung.com/spring-valid-vs-validated

Reply:
Added annotation Validated to service, NotBlank, and NotNull to Dto